aten/src/ATen/native/Copy.cpp

```diff
@@ -212,6 +212,9 @@ static Tensor & copy_impl(Tensor & self, const Tensor & src, bool non_blocking)
     return at::metal::metal_copy_(self, src);
   }
 
+  if (self.device().type() == at::kMPS || src.device().type() == at::kMPS) {
```

Comment:
Why is this code moved up? Is there an issue with transposed copy ?

Reply:
The entering condition of transpose path is `dst==cpu && src != mps/cuda/hip`, so we have no chance to use it. Moving up to prevent creating `TensorIteratorConfig`, which is unnecessary for MPS.